### PR TITLE
Prepend default PROMPT_COMMAND.

### PR DESCRIPTION
**Pull request by mikechun**
This fixes starting directory problem returning to HOME instead of current working directory when starting a new terminal tab.

### CODE REVIEW

**Comment by netj** [comment]
Could you describe more details about the problem it fixes?  Yes, preserving user's `PROMPT_COMMAND` seems like a reasonable thing to do in general, but having a concrete example would be a lot more motivating! :)  Does OS X Terminal put some special command in it for new tabs?

**Comment by mikechun** [comment]
In OS X Terminal, one of the (default) behavior for opening a new tab is "Same Working Directory".  This allows you to open new tabs in the previous `$PWD`.  However, this feature seems to be using a customized `PROMPT_COMMAND`.  If old command is not preserved, new tabs always start in `$HOME` directory regardless of the Terminal configuration.

**Comment by netj** [comment]
I don't think OS X Terminal relies on `PROMPT_COMMAND` for that feature, because I'm happily using it even without this patch!  Even for new terminal windows!  Could you check other parts of your bashrc to make sure there's no silly `cd` in them?  Or whether your shells are actually running inside `tmux` or `screen`?

One concern I have with inheriting `PROMPT_COMMAND` is the case when bpm is loaded more than once.  I should check more carefully, but I have a feeling that some bpm plugins may behave strangely if bpm installs duplicate commands into the variable.

**Comment by mikechun** [comment]
Strange.. I commented out everything but bpm and what happens is that if I turn off all plugins the issue goes away.  But if I have `prompt` plugin running then this happens.  I am not running inside `tmux` or `screen`.   I made a video below with bpm on/off.   Does your terminal behave differently?

http://screencast.com/t/zTmliMiGZU

I see your concern regarding duplicate loading, perhaps the patch needs a more sophisticated mechanism.

**Comment by netj** [comment]
I think I now know how Terminal's Same Working Dir works.  Try this as well:

``` bash
bpm on xtermtitle 
```

That plugin reflects the cwd in the terminal title/status, and perhaps that's how OS X knows where the shell is.  That piece in the plugin originates from OS X's builtin /etc/profile or bashrc, so overriding `PROMPT_COMMAND` certainly can disable the functionality.

Let's see if your patch plays nice with xtermtitle as well, and also make sure bpm doesn't install duplicate commands into it.

**Comment by mikechun** [comment]
Hey,

Sorry for super late response.  I've been overloaded recently.  I will try
your suggestions on this weekend and let you know how it goes.

Thanks always!
M

On Tue, Feb 17, 2015 at 2:42 PM, Jaeho Shin notifications@github.com
wrote:

> I think I now know how Terminal's Same Working Dir works. Try this as well:
> 
> bpm on xtermtitle
> 
> That plugin reflects the cwd in the terminal title/status, and perhaps
> that's how OS X knows where the shell is. That piece in the plugin
> originates from OS X's builtin /etc/profile or bashrc, so overriding
> PROMPT_COMMAND certainly can disable the functionality.
> 
> Let's see if your patch plays nice with xtermtitle as well, and also make
> sure bpm doesn't install duplicate commands into it.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/netj/bpm/pull/2#issuecomment-74771223.

## 

Michael Chun

File: plugin/bpm.events [hunk 1]
```diff
@@ -33,7 +33,7 @@ bash_insert_chpwd() { bash_insert_command CHPWD_COMMAND "$@"; }
 
 
 ## hooks for prompt change (PROMPT_COMMAND)
-PROMPT_COMMAND="-chpwd-detect"
+PROMPT_COMMAND="${PROMPT_COMMAND:+$PROMPT_COMMAND ;} -chpwd-detect"
```
Comment (netj) [inline plugin/bpm.events:36]:
```suggestion
PROMPT_COMMAND="${PROMPT_COMMAND:+$PROMPT_COMMAND;}-chpwd-detect"
```

This eventually got in (6f4c8b1ecce3c77c637128df776129f478996ca2) but still want to merge this PR.

**Review by netj — APPROVED** [review]

**Pull request merged**